### DISCUSSION TITLE
Trouble understanding how to play with #WP255, 254 and 0

### DISCUSSION THREAD

**Discussion by jongell** [iNavFlight/inav#9561] [Q&A] [answered]
Referencing https://github.com/iNavFlight/inav/wiki/MSP-Navigation-Messages#updating-special-waypoints

Here is what I want to do:

I want to customize the approach on specific waypoints during flight based on the wind direction (Using API to request current wind direction with a Raspberry Pi).
I'm on a fixed wing Airplane running INAV 6.1.1.

Related questions:

-As understood from the Docs I could flick the plane into Poshold and set #WP255 to a calculated Position to get the plane to fly
to, once I'm happy with the approach I could flick it back into WP and continue the mission. Is this even possible with fixed wing? ("follow me"). If yes how do I set that waypiont in terms of p1, p2, p3? 

-Could I set #WP0 during flight and that will realtime adjust the home position to that position?

If #WP254 is already the target position could I just temporarily set this to the location I want to have during waypoint mode to adjust the approach to the desired waypoint?

Thanks

**Comment by stronnag** [comment, marked as answer]
The [wiki article](https://github.com/iNavFlight/inav/wiki/INAV-Remote-Management,-Control-and-Telemetry ) contains further information on the matter, including links to example applications / code.

**Reply by jongell** [reply]
That cleared up a lot thanks! Let me allow one follow up regarding the GCS NAV Mode:

Is there any other effect once I enable the GCS NAV Mode or is it just a sanity check to enable setting WP255 and WP0 via MSP in air?

**Comment by stronnag** [comment]
> Is there any other effect once I enable the GCS NAV Mode or is it just a sanity check to enable setting WP255 and WP0 via MSP in air?

It is just a sanity "yes, I really want to do this" check.

**Reply by jongell** [reply]
Got a further question regarding the "follow me" prosedure:

Currently I'm setting #WP255 (with GCS_NAV and NAV_LOITER activated) via MSP_SET_WP with Wp number, Lat, Lng, altitude and p3 = 1 for sea level reference altitude. all the other fields I set to 0.

Specifically I set action, p1, p2 and flag to 0. Do I need to set them to specific values when sending the MSP command for this to work? Couldn't find any docs on it

**Reply by jongell** [reply]
Turns out for "follow me", one has to set via MSP_SET_WP code 209 (with NAV POSHOLD and GCS NAV activated):

WP number: 255
action: 1 (setting this to 0 will make the message invalid)
lat
lng
altitude above takeoff
p1: 0
p2: 0
p3: 0
flag: 0
